File: RegistroCarrosApp_Completo/app/src/main/java/com/example/registrocarros/network/ApiInterface.java
package com.example.registrocarros.network;

import com.example.registrocarros.model.Carro;
import com.example.registrocarros.model.Usuario;
import com.google.gson.JsonObject;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Query;
import okhttp3.ResponseBody;

public interface ApiInterface {
    // Carros
    @GET("carros.php")
    Call<List<Carro>> listarCarros(@Query("acao") String acao);

    @FormUrlEncoded
    @POST("carros.php")
    Call<JsonObject> criarCarro(
        @Field("modelo") String modelo,
        @Field("marca") String marca,
        @Field("ano") int ano,
        @Field("preco") double preco,
        @Field("descricao") String descricao,
        @Field("acao") String acao
    );

    @FormUrlEncoded
    @POST("carros.php")
    Call<JsonObject> atualizarCarro(
        @Field("id") int id,
        @Field("modelo") String modelo,
        @Field("marca") String marca,
        @Field("ano") int ano,
        @Field("preco") double preco,
        @Field("descricao") String descricao,
        @Field("acao") String acao
    );

    @FormUrlEncoded
    @POST("carros.php")
    Call<JsonObject> deletarCarro(
        @Field("id") int id,
        @Field("acao") String acao
    );

    // Usuários
    @FormUrlEncoded
    @POST("usuarios.php")
    Call<JsonObject> loginUsuario(
        @Field("email") String email,
        @Field("senha") String senha,
        @Field("acao") String acao
    );

    @FormUrlEncoded
    @POST("usuarios.php")
    Call<JsonObject> registrarUsuario(
        @Field("nome") String nome,
        @Field("email") String email,
        @Field("senha") String senha,
        @Field("acao") String acao
    );

    @FormUrlEncoded
    @POST("usuarios.php")
    Call<JsonObject> logout(@Field("acao") String acao);
}

File: RegistroCarrosApp_Completo/app/src/main/java/com/example/registrocarros/network/ApiInterface.kt
package com.example.registrocarros.network

import com.google.gson.JsonElement
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.POST

interface ApiInterface {

    @POST("carros.php")
    fun listarCarros(@Body body: Map<String, @JvmSuppressWildcards Any>): Call<JsonElement>

    @POST("carros.php")
    fun criarCarro(@Body body: Map<String, @JvmSuppressWildcards Any>): Call<JsonElement>

    @POST("carros.php")
    fun atualizarCarro(@Body body: Map<String, @JvmSuppressWildcards Any>): Call<JsonElement>

    @POST("carros.php")
    fun deletarCarro(@Body body: Map<String, @JvmSuppressWildcards Any>): Call<JsonElement>

    // Usuários
    @POST("usuarios.php")
    fun loginUsuario(@Body body: Map<String, @JvmSuppressWildcards Any>): Call<JsonElement>

    @POST("usuarios.php")
    fun registrarUsuario(@Body body: Map<String, @JvmSuppressWildcards Any>): Call<JsonElement>

    @POST("usuarios.php")
    fun logout(@Body body: Map<String, @JvmSuppressWildcards Any>): Call<JsonElement>
}
